samples/matter/thermostat/src/temperature_manager.cpp

```diff
@@ -0,0 +1,134 @@
+using namespace ::chip::DeviceLayer;
+
+constexpr EndpointId kThermostatEndpoint = 1;
+
+namespace ThermostatAttributes = chip::app::Clusters::Thermostat::Attributes;
+
+uint8_t ReturnCompleteValue(int16_t Value)
+{
+    uint8_t result = Value/100; // Convert and return only complete part of value to printable type 
+    return result;
+
+}
+
+uint8_t ReturnReminderValue(int16_t Value)
+{
+
+    uint8_t remainder = (Value%100 + 5)/10; // Converts and returns only reminder part of value to printable type. 
+                                            // This formula rounds reminder value to one significant figure
+    return remainder;
+
+}
+
+
+CHIP_ERROR TemperatureManager::Init()
+{
+    app::DataModel::Nullable<int16_t> temp;
+
+
+
+    PlatformMgr().LockChipStack();
+    ThermostatAttributes::LocalTemperature::Get(kThermostatEndpoint, temp);
+    ThermostatAttributes::OccupiedCoolingSetpoint::Get(kThermostatEndpoint, &mHeatingCelsiusSetPoint);
+    ThermostatAttributes::OccupiedHeatingSetpoint::Get(kThermostatEndpoint, &mCoolingCelsiusSetPoint);
+    ThermostatAttributes::SystemMode::Get(kThermostatEndpoint, &mThermMode);
+    PlatformMgr().UnlockChipStack();;
+
+
+    mCurrentTempCelsius     = temp.Value();
+
+
+
+    LogThermostatStatus();
+
+    return CHIP_NO_ERROR;
+}
+
+void TemperatureManager::LogThermostatStatus()
+{
+
+    LOG_INF("Thermostat Status - M:%d T:%d,%d'C H:%d,%d'C C:%d,%d'C", GetMode(), 
+                                                            ReturnCompleteValue(GetCurrentTemp()),  ReturnReminderValue(GetCurrentTemp()),
+                                                            ReturnCompleteValue(GetHeatingSetPoint()),  ReturnReminderValue(GetHeatingSetPoint()),
+                                                            ReturnCompleteValue(GetCoolingSetPoint()),  ReturnReminderValue(GetCoolingSetPoint()));
+    
+    //TODO - add thermostat ui update if available 
+
+}
+
+void TemperatureManager::AttributeChangeHandler(EndpointId endpointId, AttributeId attributeId, uint8_t * value, uint16_t size)
+{
+    switch (attributeId)
+    {
+    case ThermostatAttributes::LocalTemperature::Id: {
+        int16_t Temp = *reinterpret_cast<int16_t*>(value);
+        mCurrentTempCelsius = Temp;
+    }
+    break;
+
+    case ThermostatAttributes::OccupiedCoolingSetpoint::Id: {
+        int16_t coolingTemp = *reinterpret_cast<int16_t*>(value);
+        mCoolingCelsiusSetPoint = coolingTemp;
+        LOG_INF("Cooling TEMP: %d", mCoolingCelsiusSetPoint); 
+    }
+    break;
+
+    case ThermostatAttributes::OccupiedHeatingSetpoint::Id: {
+        int16_t heatingTemp = *reinterpret_cast<int16_t*>(value);
+        mHeatingCelsiusSetPoint = heatingTemp;
+        LOG_INF("Heating TEMP %d", mHeatingCelsiusSetPoint);
+    }
+    break;
+
+    case ThermostatAttributes::SystemMode::Id: {
+        uint8_t mode = static_cast<uint8_t>(*value);
+        if (mThermMode != mode)
+        {
+            mThermMode = mode;
```

Comment:
I'm not sure, if this check is necessary. To me, this whole case could be one-liner, like `mThermMode = *value`.